src/test-collection/formatters/tree.ts

```diff
@@ -0,0 +1,125 @@
+import path from "node:path";
+
+import type {
+    TestCollection,
+    TestDisabled,
+    FormatterTreeMainRunnable,
+    FormatterTreeTest,
+    FormatterTreeSuite,
+} from "..";
+import type { Suite, Test } from "../../types";
+
+export const format = (testCollection: TestCollection): FormatterTreeMainRunnable[] => {
+    const allSuitesById = new Map<string, FormatterTreeSuite>();
+    const allTestsById = new Map<string, FormatterTreeTest>();
+
+    testCollection.eachTest((test, browserId) => {
+        if ((test as TestDisabled).disabled) {
+            return;
+        }
+
+        if (allTestsById.has(test.id)) {
+            const treeTest = allTestsById.get(test.id)!;
+
+            if (!treeTest.browserIds.includes(browserId)) {
+                treeTest.browserIds.push(browserId);
+            }
+
+            return;
+        }
+
+        const treeTest = createTreeTest(test, browserId);
+        allTestsById.set(treeTest.id, treeTest);
+
+        collectSuites(test.parent!, treeTest, allSuitesById);
```

Comment:
here, in order to create the right tree, I move from the tests (which can be disabled) up the parent suites.